File: SWIFT/Cap05/c05ex08.swift
// c05ex08.swift

import Foundation

var RESP: Character = "S"

repeat {
    print("Entre um valor numerico: ", terminator: "")
    if let inputN = readLine(), let N = Int(inputN) {
        let R = N * 3
        print("Resultado = \(R)\n")

        print("Deseja continuar? (S/N): ", terminator: "")
        if let inputRESP = readLine(), 
           let charRESP = inputRESP.uppercased().first {
            RESP = charRESP
        }
    } else {
        print("\nEntrada invalida.\n")
        continue       
    }
    print()
} while (RESP.uppercased() == "S")

print("\nTecle <Enter> para encerrar... ", terminator: "")
_ = readLine()

File: CPP/Cap05/c05ex08.cpp
// c05ex08.cpp

#include <iostream>
#include <cctype>
using namespace std;

int main(void) {
  char RESP;
  int N, R;

  RESP = 'S';
  do {
    cout << "Entre um valor numerico: ";
    cin >> N;
    cin.ignore();

    R = N * 3;

    cout << "Resultado = " << R << endl << endl;

    cout << "Deseja continuar? (S/N): ";
    cin >> RESP;
    RESP = toupper(RESP);
    cin.ignore();

    cout << endl;
  } while (toupper(RESP) == 'S');

  cout << "\nTecle <Enter> para encerrar... ";
  cin.get();

  return 0;
}
